.github/workflows/publish.yml

```diff
@@ -0,0 +1,30 @@
+# This workflows will upload a Python Package using Twine a new when a release is created
+# For more information see: https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#publishing-to-package-registries
+
+name: Publish on PYPI
+
+on:
+  release:
+    types: [created]
```

Comment:
let's run it on every time (every commit on main and every pr)

Reply:
For the publish, we need a version tag, so we have a proper version for pip.
I would propose using tags, releases are not so nice to manage in the end.

Reply:
okay makes sense.
I'll switch to running this flow on new tags, and make sure it doesn't run unless the build flow succeeded.